### DISCUSSION TITLE
How do I enable intellisense and autocomplete?

### DISCUSSION THREAD

**Discussion by KittyBorgX** [helix-editor/helix#3942] [Q&A] [answered]
How do i enable intellisense and bring up the autocomplete menu while typing? 
I'm referring to something like this: 
![image](https://user-images.githubusercontent.com/82328083/191772106-e82b6d14-9ff9-4308-baa3-ddd1fe3645eb.png)

**Comment by the-mikedavis** [comment, marked as answer]
Language Server features are enabled out-of-the-box but you will need to install the appropriate Language Server for the language you're working in https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers

**Reply by KittyBorgX** [reply]
I've installed the `rust-analyser` binary and `hx --health` shows it as installed and active. 
![image](https://user-images.githubusercontent.com/82328083/191784463-8b9f1387-c48f-4539-8b96-88507b06b0f4.png)

Although, I haven't added anything in the `~/.config/helix/config.toml` apart from a theme choice or in the `~/.config/helix/languages.toml`

**Comment by KittyBorgX** [comment]
![image](https://user-images.githubusercontent.com/82328083/191785949-c952ccb9-07ed-4800-90f6-c6243dc5d899.png)
Update : It works now! I have no clue why, how or when. 
The link given by @the-mikedavis was really helpful! Thanks!

Tl;dr - Install the `rust-analyser` binary and you should be good to go :^)

**Reply by mandarvaze** [reply]
Here is a funny thing (I experienced it myself) - **Even if there is `~/.cargo/bin/rust-analyzer` it may not be the "correct one"**
I assume `--health` just checks for the existing of a file - hence the check passes.

Only sure way to get LSP working is `rustup component add rust-analyzer` 

Or 

confirm that `~/.cargo/bin/rust-analyzer` really works, by running `~/.cargo/bin/rust-analyzer` binary.  

If you get `rust-analyzer not found in tool-chain` (or some message like that) - you won't get LSP features from rust-analyzer with Helix. Unless you have done `rustup component add rust-analyzer`  earlier.

**Reply by kwmlodozeniec** [reply]
I had this issues and "re-installing" it this way sorted it out. Thanks!
